net/http/client.cpp

```diff
@@ -104,11 +106,11 @@ enum RoundtripStatus {
 
 class ClientImpl : public Client {
 public:
-    PooledDialer m_dialer;
+    std::unique_ptr<PooledDialer> m_dialer;
```

Comment:
why change it into heap-allocated?

Reply:
Because PooledDialer has its own unique_ptr in default constructor, copy constructor is implicitly deleted.

So no way to assign it in the member initializer list.